pandas/core/interchange/from_dataframe.py

```diff
@@ -179,9 +179,7 @@ def categorical_column_to_series(col: Column) -> tuple[pd.Series, Any]:
     if not categorical["is_dictionary"]:
         raise NotImplementedError("Non-dictionary categoricals not supported yet")
 
-    mapping = categorical["mapping"]
-    assert isinstance(mapping, dict), "Categorical mapping must be a dict"
-    categories = np.array(tuple(mapping[k] for k in sorted(mapping)))
+    categories = np.array(categorical["categories"]._col)  # type:ignore[union-attr]
```

Comment:
The `assert` above looks like it was there to help Mypy (type narrows the union to the actual type, before a `dict` and now an `np.array`). Did you try updating rather than removing it? It may remove the need for this `type:ignore` and the entry in the `pyright_` file.

Reply:
Thanks for the catch! Added back an assertion to prevent ignoring typing here.